### PR TITLE
issue: 922411 fix using fork in vma

### PR DESCRIPTION
**Pull request by rafiw**
use AC_CHECK_FUNCS_ONCE instead of AC_CHECK_LIB

Signed-off-by: Rafi Wiener <rafiw@mellanox.com>

### CODE REVIEW

**Comment by rafiw** [comment]
@miked-mellanox is this ok?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1410/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/941/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
👍

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1412/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/943/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1413/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/944/ for details (Mellanox internal link).

**Comment by ilansmith** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1414/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/945/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/946/ for details (Mellanox internal link).

File: configure.ac [hunk 1]
```diff
@@ -811,8 +811,15 @@ AC_CHECK_LIB([stdc++], [atoi])
 AC_CHECK_LIB([dl], [dlsym])
 AC_CHECK_LIB([rt], [clock_gettime])
 AC_CHECK_LIB([pthread], [pthread_create])
-#AC_CHECK_LIB([rdmacm], [rdma_create_id]) - doing it here requires rdmacm & libibverbs for udp_lat even on Ethernet only machines
-#AC_CHECK_LIB([ibverbs], [ibv_cmd_poll_cq])
+
+TMP_LDFLAGS=$LDFLAGS
+AC_CHECK_LIB([rdmacm], [rdma_create_id],
+		[LDFLAGS="$LDFLAGS -lrdmacm" VERBS_FLAGS="$VERBS_FLAGS -lrdmacm"],
+		[AC_MSG_FAILURE([rdmacm not found])])
```
Comment (ilansmith) [inline configure.ac]: @rafiw, remove the _action-if-not-found_ section both here and on line 821. It looks to me like this should fix the Jenkins failure with ICC.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1415/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/947/ for details (Mellanox internal link).

**Comment by rafiw** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1416/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1417/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/948/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1418/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/949/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1419/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1420/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/950/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1421/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/951/ for details (Mellanox internal link).

**Comment by ilansmith** [comment]
Suggested commit message:

    issue: 922411 Fix rdma_lib_reset() auto detection

**Comment by ilansmith** [comment]
Squashed and pushed to vNext.

**Comment by ilansmith** [comment]
@miked-mellanox, In the case of `AC_CHECK_LIB()` it's only added implicitly if there's no _action-if-found_  specified (which in our case it is since we're setting the `VERBS_FLAGS=...`). We specify `LDFLAGS=...` here because it's needed it for the `AC_CHECK_FUNCS_ONCE()` test.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1423/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/953/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
@ilansmith - i see, thanks.

**Comment by OphirMunk** [comment]
Merged #329

**Pull request closed without merging**